### ISSUE TITLE
Add meaningful id and class names in html tags of Camel tree items/nodes/folders

### ISSUE THREAD

**Issue by jsolovjo** [hawtio/hawtio#2826]
Hawtio v3 uses less meaningful id and class names for Camel tree items/nodes/folders as shown on the screenshot below

![image](https://user-images.githubusercontent.com/49488040/234312802-a6be1b40-0d91-4476-be67-81e9e5515bb1.png)

For example, some folders have id names like routes-9, endpoints-3, etc., some tags do not have anything at all. This approach complicates test automation a lot as a more complex solution using xpath expression (referring to parent and sibling elements) is required now which is time consuming and less reliable.

Previous Hawtio had more meaningful and logical id and class names, where you could follow what context it belongs to, easily access any folder and node by simple id/class name expression.

![image](https://user-images.githubusercontent.com/49488040/234312858-b0611c09-af7b-44ac-b756-6de7a2bfcefe.png)

Please, if possible - could you rework id and class names for Camel tree items/nodes/folders?

Thank you

**Comment by joshiraez** [comment]
I think I can bring this up quickly so I can unblock you. One sec

**Comment by joshiraez** [comment]
Also, please post this in HawtIO Next better I think

**Comment by joshiraez** [comment]
Ok, I did a quick fix for the moment. I'll add more details in the PR but for now it should be the same ids in the li elements in both versions (recursively gets name of prior elements it seems)

![Screenshot from 2023-04-25 19-06-19](https://user-images.githubusercontent.com/17336236/234351615-a1d95834-89fb-4ca7-a669-05c216857fdd.png)

This should unblock you for now but it should be refactored. I'll add more details in the PR: https://github.com/hawtio/hawtio-next/compare/main...joshiraez:camel-tree-ids?expand but you should be able to keep working

**Comment by tadayosi** [comment]
Hi @jsolovjo, thanks for reporting it.

Isn't the issue that every element in html should have some meaningful ID?  Instead of just the JMX and Camel tree nodes having them?  If so, we should rework the entire react components so that they have an ID defined.

To help us fix the issue, could you be more specific about what the e2e tests expect for html ids and classes?

**Comment by tadayosi** [comment]
So, off the top of my head, what we should do would be to add ID to every possible React component with the following format:

- `[plugin name]-[view name]-[component name](-[sub component name])`

@jsolovjo Does that work for you?

**Comment by jsolovjo** [comment]
With the changes in the PR, now I can do the following - unified the methods and reusable by other Camel tests (incl. routes, components, endpoints, contexts, etc)

Cucumber feature scenario:
```
  Scenario Outline: Check that the Attributes table under Attribute tab on Camel Specific Endpoint page is not empty
    Given User is on Camel page
    And User is on Camel "mock://result" item of "endpoints" folder of "SampleCamel" context
    When User clicks on Camel "Attributes" tab
    Then Camel table "<column>" column is not empty
    And Camel table has "<attribute>" key and "<value>" value

    Examples: Columns
    | column    | attribute       | value                  |
    | Attribute  | EndpointUri | mock://result     |
    | Value       | CamelId        | SampleCamel   |
```

One of the step definitions:
```
@And("^User is on Camel \"([^\"]*)\" item of \"([^\"]*)\" folder of \"([^\"]*)\" context$")
public void userIsOnCamelItemOfFolderOfContext(String item, String folder, String context) {
    camelPage.camelTree()
        .expandSpecificFolder(CamelTree.class, context)
        .expandSpecificFolder(CamelTree.class, context + "-" + folder)
        .selectSpecificItem(context + "-" + folder + "-" + item);
}
```
    
and the logic code behind
   
```
public <P> P expandSpecificFolder(Class<P> pageObjectClass, String folderPartialId) {
    $("[id$='" + folderPartialId + "']").$("[class$='node-toggle']").shouldBe(interactable).click();
    return page(pageObjectClass);
}
    
public void selectSpecificItem(String itemPartialId) {
    $("[id$='" + itemPartialId + "']").shouldBe(interactable).click();
 }
```

Thanks

**Comment by tadayosi** [comment]
Let us close this one as we now have useful ids for the Camel & JMX trees. If you find you need more ids for other views, please file another issue.
